language/move-stdlib/sources/vector.move

```diff
@@ -146,7 +146,26 @@ module std::vector {
         pragma intrinsic = true;
     }
 
-    /// Swap the `i`th element of the vector `v` with the last element and then pop the element.
+    /// Insert `e` at position `i` in the vector `v`, shifting the old `v[i]` and all subsequent elements to the right.
+    /// This is O(n) and preserves ordering of elements in the vector.
+    /// Aborts if `i` is out of bounds.
+    public fun insert<Element>(v: &mut vector<Element>, e: Element, i: u64) {
+        let len = length(v);
+        // i out of bounds; abort
+        if (i >= len) abort EINDEX_OUT_OF_BOUNDS;
```

Comment:
This works fine for i = len, any reason to disallow this? Might be unexpected, to be fair.

Reply:
Good question--this is certainly semantically coherent, and I actually initially implemented it this way. But I then realized that all of the other vector functions abort if `i` is not in bounds, and thus decided it was simplest to follow the same convention here.

Reply:
Conventionally in other languages (I checked Rust, C++, Python and Java) this works for `i = len` so my 2c is to also allow it here.